### ISSUE TITLE
[Android] - Still crashing when taking picture in some devices

### ISSUE THREAD

**Issue by andrepatta** [WebsiteBeaver/react-native-document-scanner-plugin#52]
Hi,

We are still experiencing some crashes when we take picture on Android, using the latest react-native-document-scanner-plugin v0.7.1 (with android-document-scanner v1.3.2 bundled) which was meant to fix this issue.

![error_img](https://user-images.githubusercontent.com/9854773/213548391-13bcfbec-0eef-4d88-8ca8-c12962d69fb1.jpeg)

Any thoughts?

Thanks!

**Comment by dmarcs** [comment]
How can I reproduce this? 0.7.1 was meant to fix a different error than this.

**Comment by gilcenio** [comment]
I'm part of the same team as @andrepatta and this is an example of the function we are using
@dmarcs example of how I'm using the function

tests performed on a moto g 10 version android 11 

```
const scanDocument = async () => {
    if (Platform.OS === 'android' && await PermissionsAndroid.request(
      PermissionsAndroid.PERMISSIONS.CAMERA
    ) !== PermissionsAndroid.RESULTS.GRANTED) {
      Alert.alert('Error', 'User must grant camera permissions to use document scanner.')
      return
    }

    const { scannedImages } = await DocumentScanner.scanDocument()

    if (scannedImages.length > 0) {
      addDocument(scannedImages[0], 'image/png')
    }
  }
```

**Comment by andrepatta** [comment]
Any updates @dmarcs ? Thanks

**Comment by dmarcs** [comment]
I tried running the code above, and I don't get that error.

**Comment by Simranjits11** [comment]
Any update?

**Comment by jrnp97** [comment]
> I tried running the code above, and I don't get that error.

I'm getting the same error, you can simulate it with the above code but enabling the `Don't keep activities` option from the developer options (at least on android)

traceback:
```
java.lang.RuntimeException: Unable to resume activity 
{<app_name>/com.websitebeaver.documentscanner.DocumentScannerActivity}: java.lang.RuntimeException: Failure delivering result ResultInfo{who=null, request=1328060520, result=-1, data=null} to activity 
{<app_name>/com.websitebeaver.documentscanner.DocumentScannerActivity}: java.lang.NullPointerException: Attempt to invoke virtual method 'int android.graphics.Bitmap.getWidth()' on a null object reference
```

**Comment by dmarcs** [comment]
Fixed in 0.9.1 by making sure documentScanner isn't null before calling handleDocumentScanIntentResult

https://github.com/WebsiteBeaver/react-native-document-scanner-plugin/pull/98

@jrnp97 Thanks for these details, but it's unrelated to this issue.
